### PR TITLE
Skip bookmark check when cancelling downloads

### LINKED ISSUE #5851: Bookmarked chapters download not being cancelled
### Steps to reproduce

1. Bookmark a chapter
2. Start to download it
3. Try to cancel it

### Expected behavior

The download should be canceled just fine.

### Actual behavior

The download isn't being canceled.

### Crash logs

_No response_

### Tachiyomi version

Preview r3756

### Android version

Android 10

### Device

Moto G7 Power

### Other details

My guess is that canceling from chapter page uses the same method as deleting, so it also checks for the "Don't delete bookmarked chapters" option.
Laziest scenario you could just overload it with an additional param if it's being canceled, which skips that check. That should work in order to change just a few pieces of code. (I think, if Kotlin also allows that)

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] I have updated the app to version **[0.12.1](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by hyoretsu**
Should close #5851 if nothing went wrong/I didn't do any mistakes.

### CODE REVIEW

**Comment by hyoretsu** [comment]
Semantics might be a little off since I'm used to TS which is pretty different from Kotlin

File: app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadManager.kt [hunk 1]
```diff
@@ -236,8 +236,13 @@ class DownloadManager(private val context: Context) {
...
      * @param source the source of the chapters.
      */
-    fun deleteChapters(chapters: List<Chapter>, manga: Manga, source: Source): List<Chapter> {
-        val filteredChapters = getChaptersToDelete(chapters)
+    fun deleteChapters(chapters: List<Chapter>, manga: Manga, source: Source, isCancelling: Boolean = false): List<Chapter> {
+        var filteredChapters = chapters
+
+        if (isCancelling == false) {
+            filteredChapters = getChaptersToDelete(chapters)
+        }
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadManager.kt:244]:
```kotlin
val filteredChapters = if (isCancelling) {
    chapters
} else {
    getChaptersToDelete(chapters)
}
```

Comment (hyoretsu) [reply app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadManager.kt:244]: Ooohh, that's closer to the ternary I originally thought of. I also wasn't sure if simply doing `(isCancelling)` was valid in Kotlin.

Comment (arkon) [reply app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadManager.kt:244]: You can just commit again to the same branch. I'll squash them when merging.

**Comment by arkon** [comment]
Thanks!

**Pull request merged**